yt/yt/server/http_proxy/solomon_proxy.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "public.h"
+#include "component_discovery.h"
+
+#include <yt/yt/library/profiling/solomon/proxy.h>
+
+namespace NYT::NHttpProxy {
+
+////////////////////////////////////////////////////////////////////////////////
+
+//! Works for all components represented in the `EYTComponentType` enum.
+class TYTComponentEndpointProvider
+    : public NProfiling::IEndpointProvider
+    , public TComponentDiscoverer
+{
+public:
+    TYTComponentEndpointProvider(
+        NApi::IClientPtr client,
+        TYTComponentEndpointProviderConfigPtr config);
+
+    TString GetComponentName() const override;
+    // TODO(achulkov2): Introduce client-side caching, either in the endpoint provider hierarchy, or in the component discoverer itself.
+    std::vector<TEndpoint> GetEndpoints() const override;
+
+private:
+    const TYTComponentEndpointProviderConfigPtr Config_;
+};
```

Comment:
Should this class be in the header? It is not used outside of `CreateSolomonProxy`

Reply:
Good catch, it shouldn't.